src/main/java/javaslang/collection/List.java

```diff
@@ -133,12 +133,12 @@
             @SuppressWarnings("unchecked")
             final List<T> list = (List<T>) elements;
             return list;
-        } else if (elements instanceof ArrayList) {
+        } else if (elements instanceof ArrayList || elements instanceof Vector) {
             @SuppressWarnings("unchecked")
-            final ArrayList<T> arrayList = (ArrayList<T>) elements;
+            final java.util.List<T> indexedList = (java.util.List<T>) elements;
```

Comment:
you could always use List.listIterator with previous() and hasPrevious() to traverse all kinds of j.u.List backwards. No need for special cases for ArrayList and Vector.